.github/workflows/build-and-test.yaml

```diff
@@ -14,9 +14,22 @@ jobs:
       - name: Checkout
         uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4
       - name: Install dependencies
-        run: python3 -m pip install tox
+        run: |
+          curl -LsSf https://astral.sh/uv/install.sh | sh
+          source $HOME/.cargo/env
+          uv venv && source .venv/bin/activate
+          uv pip install '.[dev]'
       - name: Run linters
-        run: tox -e lint
+        run: uv run tox -e lint
+      - name: Check requirements file is in sync
+        run: |
+          uv run tox -e generate-requirements
+          if ! git diff --exit-code requirements.txt; then
+            echo "requirements.txt is out of sync, please run `tox -e generate-requirements` and commit the result."
```

Comment:
Would this fail the action? Beside echoing the message, you could also add `exit 1` such that the step/check would fail

Reply:
Ah yeh, good point. I'll add that 🤦‍♂️